### PR TITLE
created resource Users with CRUD operations

### PR DESCRIPTION
**Pull request by Bohdan-Podponarovskyi**

### CODE REVIEW

File: src/users/users.controller.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  @Get(':id')
+  findOne(@Param('id') id: string) {
+    return this.usersService.findOne(+id);
+  }
+
+  @Put(':id')
+  updateFull(@Param('id') id: string, @Body() updateFullUserDto: UpdateFullUserDto) {
+    return this.usersService.updateFull(+id, updateFullUserDto);
+  }
```
Comment (GeraPashchenko) [inline src/users/users.controller.ts:29]: Better to use ParseIntPipe (ChatGPT helped me to write this explanation;))

Security: ParseIntPipe guards against invalid input and is more secure as it integrates with Nest's exception handling framework.

Readability and Maintainability: ParseIntPipe makes your intent clear—it shows that the id must be an integer. The + operator is more ambiguous and might not be as readable to others.

Consistency: ParseIntPipe keeps your controller logic consistent and clean by reducing manual validation code.

In short, while + works for simple cases, ParseIntPipe provides safer, cleaner, and more maintainable code in NestJS.

File: src/users/interfaces/user.interface.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export interface IUser {
+  firstName: string,
+  lastName: string,
+  age: number,
+  isStudent: boolean
+}
```
Comment (GeraPashchenko) [inline src/users/interfaces/user.interface.ts:6]: I forgot to add the interface to the HM, don't you mind if I copy-paste your example?

**Review by GeraPashchenko — CHANGES_REQUESTED** [review]

File: src/users/interfaces/user.interface.ts (hunk 2 above)
Comment (Bohdan-Podponarovskyi) [reply src/users/interfaces/user.interface.ts:6]: Yes of course

File: src/users/users.controller.ts (hunk 1 above)
Comment (Bohdan-Podponarovskyi) [reply src/users/users.controller.ts:29]: Thank you! will change this moment

**Comment by GeraPashchenko** [comment]
Thank you!!!
Brilliant work
